apps/src/blockly/addons/cdoFieldParameter.ts

```diff
@@ -239,31 +241,31 @@ export default class CdoFieldParameter extends GoogleBlockly.FieldVariable {
   };
 
   /**
-   * Prompt the user for a variable name or delete a variable.
-   * @param promptText description text for window prompt
-   * @param confirmButtonLabel Label of confirm button, e.g. "Rename"
-   * @param callback with parameter (text) of new name
-   * @param prompt Whether to prompt for a string value
-   * @param isDangerCancel Should cancel button has a danger type
-   * @param defaultText default input text for window prompt
+   * Prompt the user to name or delete a parameter.
+   * @param {object} options The options object.
+   * @param {string} options.promptText Description text for window prompt.
+   * @param {string} options.confirmButtonLabel Label of confirm button, e.g., "Rename".
+   * @param {function} options.callback Callback with parameter (text) of new name.
+   * @param {boolean} options.prompt Whether to prompt for a string value.
+   * @param {string} [options.defaultText] Default input text for window prompt.
    */
-  static modalDialogName = function (
-    promptText: string,
-    confirmButtonLabel: string,
-    callback: (newName: string) => void,
-    prompt: boolean,
-    isDangerCancel: boolean,
-    defaultText?: string
-  ) {
+  static parameterPrompt = function (options: {
+    promptText: string;
+    confirmButtonLabel: string;
+    callback: (newName: string) => void;
+    prompt: boolean;
+    defaultText?: string;
+  }) {
     Blockly.customSimpleDialog({
-      bodyText: promptText,
-      prompt,
-      promptPrefill: defaultText,
-      cancelText: confirmButtonLabel,
-      isDangerCancel,
+      bodyText: options.promptText,
+      prompt: options.prompt,
+      promptPrefill: options.defaultText,
+      cancelText: options.confirmButtonLabel,
+      isDangerCancel: !options.prompt,
```

Comment:
why can we make this inference?

Reply:
We basically have two options: rename (a prompt for a name) and delete (not a prompt). If it's not a prompt, we want the delete button to be red ("danger"). 
If it's a rename prompt, we want to disable closing on space, but if it's a delete we can let it do the default.

Reply:
@molly-moen I wonder if it the inferences would make more sense if I chose a different name like `isDelete` for this option?

Reply:
I pushed a change with that idea. Let me know if it feels clearer. I can also add comments. 
I was trying to avoid having multiple mutually exclusive boolean props since there's really only two scenarios we're supporting. The variable dialogs are simpler because they're both renaming prompts.

Reply:
That's much clearer! I was having a hard time figuring out the assocation between prompt and this being a delete dialog.